packages/ui/tailwind-design-system.ts

```diff
@@ -472,6 +531,20 @@ export default {
     }
   },
   plugins: [
+    /**
+     * Plugins has to be added in the order of dependency.
+     *
+     * Example: "badgeStyles" - Depends on "typographyStyles".
+     * Hence "typographyStyles" has to be added before "badgeStyles".
+     *
+     */
+    plugin(({ addUtilities }) => {
+      addUtilities(typographyStyles)
+    }),
+    plugin(({ addComponents }) => {
+      addComponents(badgeStyles)
+      addComponents(dialogStyles)
```

Comment:
For each component (e.g., Badges, Buttons, Inputs, etc.), we will maintain a separate file that contains its respective styles.